source/common/grpc/typed_async_client.cc

```diff
@@ -0,0 +1,63 @@
+#include "common/grpc/typed_async_client.h"
+
+#include "common/buffer/zero_copy_input_stream_impl.h"
+#include "common/common/utility.h"
+#include "common/grpc/common.h"
+#include "common/http/utility.h"
+
+namespace Envoy {
+namespace Grpc {
+
+void UntypedAsyncStream::sendMessageUntyped(const Protobuf::Message& request, bool end_stream) {
+  stream_->sendMessage(stream_->isGrpcHeaderRequired() ? Common::serializeBody(request)
```

Comment:
We should discuss why this is needed. I know you left some IM about this, but it's disappeared, sorry about that. I would ideally not have any special case logic here.

Reply:
This is needed because the wrappers need to serialize to the raw buffer form and that format is different depending whether or not grpc framing is required.  Yes, that was the subject of the IM; I'll try to turn on history so that things won't get lost in the future.

Reply:
As discussed, I removed isGrpcHeaderRequired().  PTAL